spec/unit/type/exec_spec.rb

```diff
@@ -408,16 +421,40 @@ def test(command, valid)
         expect(@test[param]).to eq(input)
       end
     end
+
+    describe "when given an array of arrays as input" do
+      let(:test) { Puppet::Type.type(:exec).new(:name => @executable) }
+
+      it "accepts the array when all commands return valid" do
+        input = [['one, -a'], ['two', '-b'], 'three']
+        expect(test.provider).to receive(:validatecmd).exactly(input.length).times.and_return(true)
+        test[param] = input
+        expect(test[param]).to eq(input)
+      end
+
+      it "rejects the array when any commands return invalid" do
+        input = [['one, -a'], ['two', '-b'], 'three']
+        expect(test.provider).to receive(:validatecmd).with(input.first).and_return(false)
+        input[1..-1].each do |cmd|
+          expect(test.provider).to receive(:validatecmd).with(cmd).and_return(true)
+        end
```

Comment:
Can also be written as
`expect(test.provider).to receive(:validatecmd).exactly(input.length).times.and_return(false, true, true)`

but I'm not 100% sure

Reply:
We won't be able to have `.with(cmd)` check
I'm inclined to keep them as are now since those tests are similar to the others in this file